pkg/templates/nstemplatetiers/nstemplatetier_generator.go

```diff
@@ -109,14 +110,19 @@ func parseAllRevisions(metadata []byte) (map[string]map[string]string, error) {
 		tierKind := data[0]
 		nsKind := data[1]
 		// create a new entry if needed
-		r, ok := revision.(string)
-		if !ok {
-			return nil, errors.Errorf("invalid namespace template filename revision for '%[1]s'. Expected a string, got a %[2]T ('%[2]v')", filename, revision)
+		var rev string
+		switch r := revision.(type) {
```

Comment:
Can we just fix our metadata.yaml generation so it uses SHAs as strings?
```
advanced-code: "1234567"
```
Then we can unmarshal it to `map[string]string`

Reply:
I guess we could do that, but when we need to ensure that the generated file would have this format, and this will only fail when a commit hash is a number. So I'd rather ensure that we support both cases (alphanum or nums) at this level.